tests/actor/_power_managing/test_matryoshka.py

```diff
@@ -0,0 +1,162 @@
+from frequenz.sdk.actor._power_managing import Proposal
+from frequenz.sdk.actor._power_managing._matryoshka import Matryoshka
+from frequenz.sdk.timeseries import Power, battery_pool
+
+
+async def test_matryoshka_algorithm() -> None:  # pylint: disable=too-many-statements
+    """Tests for the power managing actor."""
+    batteries = frozenset({2, 5})
+
+    algorithm = Matryoshka()
+    system_bounds = battery_pool.PowerMetrics(
+        timestamp=datetime.now(tz=timezone.utc),
+        inclusion_bounds=timeseries.Bounds(
+            lower=Power.from_watts(-200.0), upper=Power.from_watts(200.0)
+        ),
+        exclusion_bounds=timeseries.Bounds(lower=Power.zero(), upper=Power.zero()),
+    )
+
+    call_count = 0
+
+    def test_tgt_power(
+        priority: int,
+        power: float | None,
+        bounds: tuple[float | None, float | None],
+        expected: float | None,
+        must_send: bool = False,
+    ) -> None:
+        nonlocal call_count
+        call_count += 1
+        tgt_power = algorithm.get_target_power(
+            batteries,
+            Proposal(
+                battery_ids=batteries,
+                source_id=f"actor-{priority}",
+                preferred_power=None if power is None else Power.from_watts(power),
+                bounds=timeseries.Bounds(
+                    None if bounds[0] is None else Power.from_watts(bounds[0]),
+                    None if bounds[1] is None else Power.from_watts(bounds[1]),
+                ),
+                priority=priority,
+            ),
+            system_bounds,
+            must_send,
+        )
+        assert tgt_power == (
+            Power.from_watts(expected) if expected is not None else None
+        )
+
+    def test_bounds(
+        priority: int,
+        expected_power: float | None,
+        expected_bounds: tuple[float, float],
+    ) -> None:
+        report = algorithm.get_status(batteries, priority, system_bounds, None)
+        if expected_power is None:
+            assert report.target_power is None
+        else:
+            assert report.target_power is not None
+            assert report.target_power.as_watts() == expected_power
+        assert report.inclusion_bounds is not None
+        assert report.inclusion_bounds.lower.as_watts() == expected_bounds[0]
+        assert report.inclusion_bounds.upper.as_watts() == expected_bounds[1]
+
+    test_tgt_power(priority=2, power=25.0, bounds=(25.0, 50.0), expected=25.0)
+    test_bounds(priority=2, expected_power=25.0, expected_bounds=(-200.0, 200.0))
+    test_bounds(priority=1, expected_power=25.0, expected_bounds=(25.0, 50.0))
+
+    test_tgt_power(priority=1, power=20.0, bounds=(20.0, 50.0), expected=None)
+    test_tgt_power(
+        priority=1, power=20.0, bounds=(20.0, 50.0), expected=25.0, must_send=True
+    )
+    test_bounds(priority=1, expected_power=25.0, expected_bounds=(25.0, 50.0))
+
+    test_tgt_power(priority=3, power=10.0, bounds=(10.0, 15.0), expected=15.0)
```

Comment:
Ha! This is confusing, I think it makes sense but it makes you think about priorities a bit differently from what one might expect. So high priority means high priority for bounds but low priority for target power, right? So this is a higher priority request so the bounds are narrowed but the actual target power is the closest to the lowest priority request instead, right? And this is why this high prio request gets `15` instead of `10`, which is what it actually wanted?

Reply:
Right.  If that's not the case, and we just prioritize preferred powers of higher pri actors, then lower pri actors might as well not bother sending any request, and we don't need bounds,  and probably no power manager, and actually the lower pri actors don't need to run as well. :D